apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -402,63 +413,41 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
     :ok
   end
 
-  def retrieve_device_stats_and_introspection!(db_client, device_id) do
-    stats_and_introspection_statement = """
-    SELECT total_received_msgs, total_received_bytes, introspection,
-           exchanged_bytes_by_interface, exchanged_msgs_by_interface
-    FROM devices
-    WHERE device_id=:device_id
-    """
-
-    device_row_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(stats_and_introspection_statement)
-      |> DatabaseQuery.put(:device_id, device_id)
-      |> DatabaseQuery.consistency(:local_quorum)
-
-    device_row =
-      DatabaseQuery.call!(db_client, device_row_query)
-      |> DatabaseResult.head()
+  def retrieve_device_stats_and_introspection(realm, device_id) do
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
```

Comment:
This logic is reused a lot, maybe a centralized function could be helpful, [astarte_data_access#71   astarte/realm.ex](https://github.com/noaccOS/astarte_data_access/blob/ae7573b0183a66090c8a091044a0a9b2e3170773/lib/astarte_data_access/astarte/realm.ex#L19-L34) has a good solution that could come in handy here too